snuba/consumers/consumer_builder.py

```diff
@@ -61,6 +62,19 @@ class MockParameters:
     std_deviation: int
 
 
+def validate_passed_slice(
+    storage_key: StorageKey, slice_id: Optional[int] = None
```

Comment:
nit: I think it's better to omit the None default here. It feels safer to force the caller to pass the slice ID. Otherwise we might risk applying the wrong default. There's a few other functions in this diff that do a similar thing too.